python/ray/services.py

```diff
@@ -957,12 +957,72 @@ def start_raylet(redis_address,
     return raylet_name
 
 
+def determine_plasma_store_config(object_store_memory=None,
+                                  plasma_directory=None,
+                                  huge_pages=False):
+    """Figure out how to configure the plasma object store.
+
+    Args:
+        object_store_memory: The user-specified object store memory parameter.
+        plasma_directory: The user-specified plasma directory parameter.
+        huge_pages: The user-specified huge pages parameter.
+
+    Returns:
+        A tuple of the object store memory to use and the plasma directory to
+            use. If either of these values is specified by the user, then that
+            value will be preserved.
+    """
+    system_memory = ray.utils.get_system_memory()
+
+    # Choose a default object store size.
+    if object_store_memory is None:
+        object_store_memory = int(system_memory * 0.4)
+
+    if plasma_directory is not None:
+        plasma_directory = os.path.abspath(plasma_directory)
+
+    # Determine which directory to use. By default, use /tmp on MacOS and
+    # /dev/shm on Linux, unless the shared-memory file system is too small,
+    # in which case we default to /tmp on Linux.
+    if plasma_directory is None:
+        if sys.platform == "linux" or sys.platform == "linux2":
+            shm_avail = ray.utils.get_shared_memory_bytes()
+            # Compare the requested memory size to the memory available in
+            # /dev/shm.
+            if shm_avail > object_store_memory:
+                plasma_directory = "/dev/shm"
+            else:
+                plasma_directory = "/tmp"
+                logger.warning(
+                    "WARNING: The object store is using /tmp instead of "
+                    "/dev/shm because /dev/shm has only {} bytes available. "
+                    "This may slow down performance! You may be able to free "
+                    "up space by deleting files in /dev/shm or terminating "
+                    "any running plasma_store_server processes. If you are "
+                    "inside a Docker container, you may need to pass an "
+                    "argument with the flag '--shm-size' to 'docker run'."
+                    .format(shm_avail))
+        else:
+            plasma_directory = "/tmp"
```

Comment:
wait this will default to `/tmp` for mac?

Reply:
Yeah, that is the behavior before and after this PR. `/dev/shm` doesn't exist on MacOS I think.